### PR TITLE
Defang Pulumi Provider Docs

### LINKED ISSUE #131: Document Defang Pulumi Provider
Add context to Pulumi provider docs page:
* Why (what are the scenarios) would I Deploy using Pulumi?
* Are there any limitations or best practices to keep in mind with this approach?

### PR DESCRIPTION
**Pull request by jordanstephens**
Closes #131 

Starting to address the issues in #131 with this PR, though I think there is still room for improvement.

I think the main issues that I want to clarify revolve around documenting the limitations of the compose file.

### CODE REVIEW

File: docs/concepts/pulumi.md [hunk 1]
```diff
@@ -13,13 +13,11 @@ import TabItem from '@theme/TabItem';
 Pulumi support is currently only available for Defang Playground. We are working on support for Defang BYOC.
 :::
 
-You can use Pulumi to define your Defang services. This allows you to integrate your Defang services with other cloud resources.
+[Pulumi](https://www.pulumi.com) is a modern infrastructure-as-code toolkit that allows developers to use a programming language like Typescript to provision and manage cloud resources.
 
-For example, you might define an Aiven PostgreSQL database in your Pulumi code and then use the outputs of that resource to configure the secrets to connect to the database in your Defang service.
+Defang providers a Pulumi [_Provider_](https://www.pulumi.com/docs/iac/concepts/resources/providers/) written in Typescript which can be used to deploy Defang services alongside other pulumi-managed infrastructure.
```
Comment (commit111) [inline docs/concepts/pulumi.md:18]:
```suggestion
Defang provides a Pulumi [_Provider_](https://www.pulumi.com/docs/iac/concepts/resources/providers/) written in Typescript which can be used to deploy Defang services alongside other Pulumi-managed infrastructure.
```

File: docs/concepts/pulumi.md [hunk 2]
```diff
@@ -47,8 +45,15 @@ yarn add @defang-io/pulumi-defang
   </TabItem>
 </Tabs>
 
+## When to Use the Defang Pulumi Provider
+
+The Defang Pulumi Provider supports developer's with more complex requirements than a [Compose File](/docs/concepts/compose) will support.
```
Comment (commit111) [inline docs/concepts/pulumi.md:50]:
```suggestion
The Defang Pulumi Provider is a good option for developers with more complex requirements than what can be currently supported by a [Compose File](/docs/concepts/compose).
```

File: docs/tutorials/deploy-using-pulumi.mdx [hunk 3]
```diff
@@ -5,33 +5,41 @@ sidebar_position: 400
...
+* [A Defang Account](/docs/concepts/authentication)
+* [Pulumi CLI](https://www.pulumi.com/docs/install)
+* [The Defang CLI](/docs/getting-started#install-the-defang-cli)
+* [The Defang Pulumi Provider](/docs/concepts/pulumi#install-the-defang-pulumi-provider)
 
-## Project Directory Setup
+## Authenticate With Defang
 
-Create a new directory for your project and navigate to it.
+Make sure you are logged into  [Defang CLI](/docs/concepts/authentication.md). Don't worry about the Pulumi CLI for now.
```
Comment (commit111) [inline docs/tutorials/deploy-using-pulumi.mdx:25]:
```suggestion
Make sure you are logged into the [Defang CLI](/docs/concepts/authentication.md). Don't worry about the Pulumi CLI for now.
```

**Review by commit111 — APPROVED** [review]
added a few typo suggestions. lgtm!

File: docs/tutorials/deploy-using-pulumi.mdx [hunk 4]
```diff
@@ -5,33 +5,41 @@ sidebar_position: 400
...
 # Deploy Using Pulumi
 
+:::warning
+At the time of writing, the Defang Pulumi Provider only works with [Defang Playground](../concepts/defang-playground.md). We are working on [BYOC](../concepts/defang-byoc.md) support.
+:::
+
 <iframe width="560" height="315" src="https://www.youtube.com/embed/A4uiP5FIgg0?si=BHZLOxN4noJ56f7x" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
 
-## Installing Dependencies
+This tutorial will show you how to deploy minio with Pulumi using the Defang Provider.
```
Comment (commit111) [inline docs/tutorials/deploy-using-pulumi.mdx:14]: can you move this sentence above the video?

**Review by raphaeltm — APPROVED** [review]

**Pull request merged**